src/main/java/frc/robot/subsystems/SwerveDriveSubsystem.java

```diff
@@ -138,9 +138,11 @@ public Command toggleFieldCentric() {
         });
     }
 
-    public void resetFieldCentric() {
-        odometry.zeroHeading();
-        resetLockRot();
+    public Command resetFieldCentric() {
+        return runOnce(() -> {
```

Comment:
`runOnce` actually requires the subsystem by default, since this is the `SubsystemBase` version. To not require it, use `Commands.runOnce`